draft/phoenix_presence_with_live_view.md

```diff
@@ -0,0 +1,141 @@
+      RealWorldPhoenix.Presence # <= Add This!
+    ]
+
+    opts = [strategy: :one_for_one, name: RealWorldPhoenix.Supervisor]
+    Supervisor.start_link(children, opts)
+  end
+
+  def config_change(changed, _new, removed) do
+    RealWorldPhoenixWeb.Endpoint.config_change(changed, removed)
+    :ok
+  end
+end
+```
+
+## I don't have any Channels... ?
+
+So the next and last step in the guide shows an example of adding Presence tracking to a channel implementation. But we are not using channels! So how do we go about and add this to our LiveView. Presence uses a PubSub (Publish-Subscribe) mechanism to perform it's magic. So what we should do in our LiveView is make sure we subscribe to the topic of the current page. Once we do that we can add a callback that will track any presence_diff where we will update the counter if someone leaves or joins the current topic.
+
+In our `mount/2` function we'll subscribe to the page topic. Basically using a topic such as `blog:simple_phoenix_presence`. So in essence a general category with a post slug after the colon. Here are the updates to our LiveView module:
+
+```elixir
+defmodule RealWorldPhoenixWeb.Live.ReaderCount do
+  ...
+
+  @topic "blog:simple_phoenix_presence"
+
+  def mount(_session, socket) do
+    # before subscribing, let's get the current_reader_count
+    initial_count = Presence.list(@topic) |> map_size
+
+    # Subscribe to the topic
+    RealWorldPhoenixWeb.Endpoint.subscribe(@topic)
+
+    # Track changes to the topic
+    Presence.track(
+      self(),
+      @topic,
+      socket.id,
+      %{}
+    )
+
+    {:ok, assign(socket, :reader_count, initial_count)}
+  end
+  # ...
+end
+```
+
+## Track diff of topic
+
+Now the only left for us to do is implement a callback in our live_view to catch any diffs happening on th etopic we are tracking. The folling function will do the trick:
+
+```elixir
+...
+  def handle_info(
+        %{event: "presence_diff", payload: %{joins: joins, leaves: leaves}},
+        %{assigns: %{reader_count: count}} = socket
+      ) do
+    reader_count = count + map_size(joins) - map_size(leaves)
+
+    {:noreply, assign(socket, :reader_count, reader_count)}
+  end
+...
+```
+
+We pattern match on the "presence_diff" event and also use pattern matching to get the current values for joins, leaves and count respectively. When someone subscribes to the topic, there will be change in the `joins` and when they leave a similar change in `leaves`. The `reader_count` is something we update ourselves in the socket whenever someone joins or leaves.
+
+That is really all we need to do to get realtime updates of the number of readers of our page. That is quite easy and really cool to be able to see that data. Currently we don't ask user for information once they enter our blog, but once we add login or even just their name, we would be able to show that data as part of the presence_diff as well. But, let's leave that for another time.
+
+## Conclusion
```

Comment:
Maybe it would be nice to show how you embed this LiveComponent in the blog post too? Feels like it's missing now.

Reply:
I have added that part to the beginning of the post now. Do you think that is the reight spot?